### PR TITLE
Cb 3115 add new create delete project events

### PR DESCRIPTION
**Pull request by EmYar**

### CODE REVIEW

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebSession.java [hunk 1]
```diff
@@ -1048,7 +1064,7 @@ public void subTask(String name) {
         }
     }
 
-    private class TaskProgressMonitor extends ProxyProgressMonitor {
+    private static class TaskProgressMonitor extends ProxyProgressMonitor {
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebSession.java:1067]: redundant change

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebSession.java [hunk 2]
```diff
@@ -1070,7 +1086,7 @@ public void subTask(String name) {
         }
     }
 
-    private class PersistentAttribute {
+    private static class PersistentAttribute {
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebSession.java:1089]: redundant change

File: server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WsProjectLifeCycleEventHandler.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import io.cloudbeaver.DBWConstants;
+import io.cloudbeaver.model.session.BaseWebSession;
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.model.websocket.event.WSEventTopic;
+import org.jkiss.dbeaver.model.websocket.event.WSEventType;
+import org.jkiss.dbeaver.model.websocket.event.WSProjectUpdateEvent;
+
+public class WsProjectLifeCycleEventHandler extends WSProjectUpdatedEventHandler<WSProjectUpdateEvent> {
```
Comment (alexander-skoblikov) [inline server/bundles/io.cloudbeaver.server/src/io/cloudbeaver/server/events/WsProjectLifeCycleEventHandler.java:12]: WsProjectLifeCycleEventHandler rename to WSProjectUpdatedEventHandler

old WSProjectUpdatedEventHandler rename to WSAbstractProjectEventHandler

File: server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebSession.java (hunk 2 above)
Comment (EmYar) [reply server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/session/WebSession.java:1089]: Static inner classes is more memory efficient. (https://stackoverflow.com/questions/20380600/gc-performance-hit-for-inner-class-vs-static-nested-class)

The IDEA has an inspection that reports the possibility of making the inner class static. Here is description
> A static inner class does not keep an implicit reference to its enclosing instance. This prevents a common cause of memory leaks and uses less memory per instance of the class.

Are there specific reasons for leaving a non-static declaration?

**Comment by EmYar** [comment]
merged into CB-2888-project-sync in commit https://github.com/dbeaver/cloudbeaver/commit/d37a7f840c65e062c4723c64af501d5ee332d9bb

**Pull request closed without merging**